ContentPatcher/ModEntry.cs

```diff
@@ -463,5 +492,71 @@ select value.Trim()
             );
             return new InvariantHashSet(values);
         }
+        
+        /// <summary>
+        /// Register a config menu field with a given GMCM integration.
+        /// </summary>
+        /// <param name="configMenuIntegration">The GMCM integration.</param>
+        /// <param name="configName">The config field name.</param>
+        /// <param name="configField">The config field itself.</param>
+        /// <param name="valueProvider">The value provider for the config token.</param>
+        private void RegisterConfigMenuField(GenericModConfigMenuIntegration<InvariantDictionary<ConfigField>> configMenuIntegration, string configName, ConfigField configField, ManualValueProvider valueProvider)
+        {
+            if (configField.AllowValues.Any())
+            {
+                if (configField.AllowMultiple)
+                {
+                    // Whitelist + multiple options = fake with multiple checkboxes
+                    foreach (string allowedValue in configField.AllowValues)
+                    {
+                        Func<InvariantDictionary<ConfigField>, bool> get = (config) => configField.Value.Contains(allowedValue);
+                        Action<InvariantDictionary<ConfigField>, bool> set = (config, val) =>
+                        {
+                            InvariantHashSet values = new InvariantHashSet(configField.Value);
+                            if (val && !values.Contains(allowedValue))
+                                values.Add(allowedValue);
+                            if (!val && values.Contains(allowedValue)) 
+                                values.Remove(allowedValue);
+
+                            if (values.Count == 0 && !configField.AllowBlank)
+                                values = configField.DefaultValues;
+
+                            configField.Value = values;
+                            valueProvider.SetValue(values);
+                        };
+                        configMenuIntegration.AddCheckbox($"{configName}.{allowedValue}", null, get, set);
+                    }
+                }
+                else
+                {
+                    // Whitelist + single value = drop down
+                    // Need an extra option when blank is allowed
+                    List<string> choices = new List<string>(configField.AllowValues);
+                    if (configField.AllowBlank)
+                        choices.Insert(0, "");
+
+                    Func<InvariantDictionary<ConfigField>, string> get = (config) => configField.Value.Count > 0 ? configField.Value.First() : "";
```

Comment:
This can be simplified:
```cs
// current code
configField.Value.Count > 0 ? configField.Value.First() : ""

// suggested
configField.Value?.FirstOrDefault() ?? ""
```

Reply:
Done.